BeautifulSoupScript_edits.py

```diff
@@ -0,0 +1,213 @@
+from bs4 import BeautifulSoup
+from bs4.element import Tag
+import requests, json
+from requests.exceptions import ConnectionError, HTTPError
+
+
+def fetchPageHTML(url):
+    elemResponse = requests.get(url)
+    elemResponse.raise_for_status()
+    return elemResponse
+
+#Array of each catalog publications
+request_urls = [
+    'https://oi.uchicago.edu/research/publications/assyriological-studies', 
+    'https://oi.uchicago.edu/research/publications/assyrian-dictionary-oriental-institute-university-chicago-cad',
+    'https://oi.uchicago.edu/research/publications/demotic-dictionary-oriental-institute-university-chicago',
+    'https://oi.uchicago.edu/research/publications/hittite-dictionary-oriental-institute-university-chicago-chd',
+    'https://oi.uchicago.edu/research/publications/chicago-hittite-dictionary-supplements-chds',
+    'https://oi.uchicago.edu/research/publications/late-antique-and-medieval-islamic-near-east-lamine',
+    'https://oi.uchicago.edu/research/publications/materials-assyrian-dictionary',
+    'https://oi.uchicago.edu/research/publications/materials-and-studies-kassite-history-mskh',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-communications-oic',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-digital-archives-oida',
+    'https://oi.uchicago.edu/research/oimp',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-nubian-expedition-oine',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-publications-oip',
+    'https://oi.uchicago.edu/research/publications/oriental-institute-seminars-ois',
+    'https://oi.uchicago.edu/research/publications/studies-ancient-oriental-civilization-saoc',
+    'https://oi.uchicago.edu/research/publications/lost-egypt',
+    'https://oi.uchicago.edu/research/publications/miscellaneous-publications'
+]
+
+
+OIC_ROOT = 'https://oi.uchicago.edu'
+
+def main(journal_key=0):
+    #Looping through each catalog to add the download links and metadata to a dictionary
+    for i, url in enumerate(request_urls):
+        try:
+            if i == 2:
+                continue
+            else:
+                elem = fetchPageHTML(url)
+        except Exception as e:
+            print(e)
+            continue
+
+        soup = BeautifulSoup(elem.text, 'lxml')   #Parsing the catalog's webpage using BeautifulSoup
+
+        #Parsing for dictionary catalogs
+        if i == 1 or i == 3 or i == 15:
+
+            catContainer = soup.find(class_='catalog')
+
+            pubList = catContainer.find('tbody')
+
+            pubEntry = pubList.find_all('tr')
+
+            for pub in pubEntry:
+                print(url)
+
+                print(parse_Pub_Dict(pub))
```

Comment:
This is a good separation of concerns! Maintaining these parsing methods helps keep this code easier to read/follow